react-state/my-app/src/ToggleButton.js

```diff
@@ -0,0 +1,23 @@
+import { useState } from "react"
+
+export default function ToggleButton({text, color}) {
+  const [isClicked, setIsClicked] = useState(false)
+
+  function handleClick() {
+    console.log('Value before calling the setter: ', isClicked)
+    setIsClicked(!isClicked)
+    console.log('Value after calling the setter: ', isClicked)
+  }
+
+  console.log('Value returned by useState: ', isClicked)
+
+  return(
+    isClicked ? (<button style={{background: color}} onClick={handleClick}>
+      {text}
+    </button>)
+    :
+    (<button style={{background: 'white'}} onClick={handleClick}>
+      {text}
+    </button>)
+  )
```

Comment:
This is really ugly and hard to read. I would simplify this a bit. The only thing that changes between these two is the background, so we should just move the ternary there.

```
return (
    <button onClick={handleClick} style={{ backgroundColor: isClicked ? color : 'white' }}>
      {text}
    </button>
  );
  ```